stacks-common/src/util/vrf.rs

```diff
@@ -533,16 +529,16 @@ impl VRF {
         let (Y_point, x_scalar, trunc_hash) = VRF::expand_privkey(secret);
         let H_point = VRF::hash_to_curve(&Y_point, alpha);
 
-        let Gamma_point = &x_scalar * &H_point;
+        let Gamma_point = x_scalar * H_point;
```

Comment:
We'll want to check the code, but the reason these are `&`-referenced might be because `ed25519-dalek` implements `Deref` for these variables, and we're really interested in getting the inner type's behavior.  I'm leery of changing this module to begin with.

Reply:
Checking these types, it does not implement Deref for these variables. This most likely is why clippy reports a needless ref

Reply:
Rethinking this because I was talking with Joey and he pointed out that this will introduce a copy that could have non neligable perofmrance impacts. I am going to silence this warning out of an abundance of caution as I have not bench marked this change.

Reply:
Joey reminded me that these structs may be actually quite large and that removing these references may have non negligable performance impacts (for the mathy bits). Since I have not benchmarked this change, i have reverted and silenced the warning with a clippy allow out of an abundance of caution.